onnxruntime/core/providers/qnn/builder/opbuilder/slice_op_builder.cc

```diff
@@ -75,6 +72,45 @@ void SliceOpBuilder::GetDataFromAttribute(const NodeUnit& node_unit,
   }
 }
 
+// Gets the data from initializer inputs (e.g., starts, ends, axes, or steps) as a TensorShapeVector.
+static Status GetInitializerInputData(const NodeUnitIODef& input, const QnnModelWrapper& qnn_model_wrapper,
+                                      TensorShapeVector& output) {
+  OnnxInputInfo input_info = {};
+  ORT_RETURN_IF_ERROR(qnn_model_wrapper.GetOnnxInputInfo(input, input_info));
+  ORT_RETURN_IF_NOT(input_info.is_initializer,
+                    "QNN requires the starts, ends, axes, and steps inputs to "
+                    "be initializers");
+
+  std::vector<uint8_t> initializer_bytes;
+  ORT_RETURN_IF_ERROR(qnn_model_wrapper.UnpackInitializerData(*input_info.initializer_tensor, initializer_bytes));
+
+  size_t tensor_byte_size = initializer_bytes.size();
+  const auto data_type = input_info.initializer_tensor->data_type();
+
+  Status status;
+
+  switch (data_type) {
+    case ONNX_NAMESPACE::TensorProto_DataType_INT64: {
+      const int64_t* tensor_data = reinterpret_cast<const int64_t*>(initializer_bytes.data());
+      size_t size = tensor_byte_size / sizeof(int64_t);
+      output.insert(output.end(), tensor_data, tensor_data + size);
+      break;
+    }
+    case ONNX_NAMESPACE::TensorProto_DataType_INT32: {
+      const int32_t* tensor_data = reinterpret_cast<const int32_t*>(initializer_bytes.data());
```

Comment:
Is it gauranteed that data properly aligned and contigious?

Reply:
Yes, I added the following comment (above) to explain:

```C++
// UnpackInitializerData() uses ORT's protobuf utilities, which ensure that the initializer bytes are
// contiguous, aligned, and in the appropriate endianness. This is necessary to be able to reinterpret bytes
// as an array of larger elements.
```

Reply:
The `initializer_bytes` variable is just a normal vector, which uses the default malloc/free to manage memory.

Reply:
Yes, the contents of `initializer_bytes` are copied to a different output vector before it goes out of scope.